### PR TITLE
Adding filtering to the World Hopper Plugin

### LINKED ISSUE #4970: Sort by member's world and country - World Hopper
Add ability to sort world based on member-only and country similar to the vanilla world hopper.

### PR DESCRIPTION
**Pull request by RoyceMathews**
Fixes #4970 - Adding filtering to the World Hopper Plugin

* Adds a new button above the header used to sort the worlds. The button expands a panel with checkboxes for selecting filters.
* Checks WorldTypes to determine which worlds to display.
* `WorldHopperPlugin#updateList` is used since I wanted to update the list of worlds without triggering a world refresh that may be throttled.

### CODE REVIEW

**Comment by raiyni** [comment]
Need a screenshot

**Comment by RoyceMathews** [comment]
capture1
capture

**Comment by RoyceMathews** [comment]
Note from Discord:

Should add a filter icon to the Activity Header and have the checkboxes show up in a popup menu.

**Comment by RoyceMathews** [comment]
capture
Thoughts on placement in the activity header?

**Comment by RoyceMathews** [comment]
capture
The gear to the left of the arrow.

**Comment by Dgc2002** [comment]
Would it be possible to add a filter icon for better UX?  
  
Like [this one](https://commons.wikimedia.org/wiki/File:Toicon-icon-lines-and-angles-filter.svg) for example.  
  
Also would it be worth your while to add filter options to the World and Ping columns as well?  
  
World would have the Members/Free filters. Maybe allow filtering based on server location(`locationId`)?  
Ping would allow you to filter worlds above a ping threshold(e.g. "Hide servers with pings higher than X").

**Comment by raiyni** [comment]
I think the funnel icon is dumb

**Comment by Dgc2002** [comment]
@raiyni That's fine but it's also the standard way to indicate filtering. It's immediately recognizable and clearly communicates functionality.

**Comment by RoyceMathews** [comment]
@Dgc2002 I kind of like being able to access the filters from one place vs having to to click on several filters.

**Comment by Dgc2002** [comment]
@RoyceMathews Yea I can see that working as well. Though the additional filters I mentioned would still be nice.

**Comment by kg-89** [comment]
Is it possible to add the filter to the ingame world hopper too?

**Comment by JaydenMatas** [comment]
Has the ability to sort by country also been added? If not, I would like to have a go at implementing this if that is okay.

File: runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/filter/WorldFilter.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	public WorldFilter(WorldFilterDefinition aWorldFilterDefinition)
+	{
+		worldFilterDefinition = aWorldFilterDefinition;
+	}
+
+	public boolean hide(Collection<WorldType> aWorldTypes)
+	{
+		if (worldFilterDefinition.shouldHideF2P())
+		{
+			return !aWorldTypes.contains(WorldType.MEMBERS)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/filter/WorldFilter.java]: Isn't this doing the opposite?

**Comment by Adam-** [comment]
Is there a reason there is so much indirection with this hide logic? I think it can be done with just a switch?

**Comment by ShaunDreclin** [comment]
Using these filters breaks the ping display for some reason.

screenshot

**Comment by al3x-huang** [comment]
Any word on this?

**Comment by Nightfirecat** [comment]
This PR has been inactive for some time, pending a number of updates, so I am closing it for now.

RoyceMathews, if you would like to make further updates to this PR, please re-open it and push whatever changes you want to make.

**Pull request closed without merging**